src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -224,47 +224,92 @@ private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<byt
 #endif
 
             byte[] output = new byte[estimatedSize];
-            int numBytesNeeded;
-            ErrorCode errorCode = encrypt ?
-                Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode) :
-                Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+            int numBytesNeeded = 0;
+
+            ErrorCode errorCode = 0;
+
+            for (int i = 0; i <= StatusUnsuccessfulRetryCount; i++)
+            {
+                errorCode =
+                    EncryptOrDecrypt(key, input, output, paddingMode, paddingInfo, encrypt, out numBytesNeeded);
+
+                if (errorCode != ErrorCode.STATUS_UNSUCCESSFUL)
+                {
+                    break;
+                }
+            }
 
             if (errorCode == ErrorCode.NTE_BUFFER_TOO_SMALL)
             {
                 output = new byte[numBytesNeeded];
-                errorCode = encrypt ?
-                    Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode) :
-                    Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
 
+                for (int i = 0; i <= StatusUnsuccessfulRetryCount; i++)
+                {
+                    errorCode =
+                        EncryptOrDecrypt(key, input, output, paddingMode, paddingInfo, encrypt, out numBytesNeeded);
+
+                    if (errorCode != ErrorCode.STATUS_UNSUCCESSFUL)
+                    {
+                        break;
+                    }
+                }
             }
+
             if (errorCode != ErrorCode.ERROR_SUCCESS)
             {
                 throw errorCode.ToCryptographicException();
             }
 
-            Array.Resize(ref output, numBytesNeeded);
+            if (numBytesNeeded != output.Length)
+            {
+                Span<byte> writtenSpan = output.AsSpan(0, numBytesNeeded);
+                byte[] ret = writtenSpan.ToArray();
+                CryptographicOperations.ZeroMemory(writtenSpan);
+                output = ret;
+            }
```

Comment:
This doesn't look related to this fix.  You're doing this to be able to clear the original buffer?

Reply:
Yeah, it was a "grr, this is wrong and in a method I'm fixing".  It looks like it broke UWP somehow, so I'm going to undo this part and just do it as a separate change.